frontend/src/pages/modelServing/screens/projects/NIMServiceModal/ManageNIMServingModal.tsx

```diff
@@ -222,15 +231,24 @@ const DeployNIMServiceModal: React.FC<DeployNIMServiceModalProps> = ({
       submitServingRuntimeResources({ dryRun: true }),
       submitInferenceServiceResource({ dryRun: true }),
     ])
-      .then(() =>
-        Promise.all([
-          submitServingRuntimeResources({ dryRun: false }),
-          submitInferenceServiceResource({ dryRun: false }),
-          createNIMSecret(namespace, NIM_SECRET_NAME, false, false),
-          createNIMSecret(namespace, NIM_NGC_SECRET_NAME, true, false),
-          createNIMPVC(namespace, nimPVCName, pvcSize, false),
-        ]),
-      )
+      .then(async () => {
+        const promises: Promise<void>[] = [
+          submitServingRuntimeResources({ dryRun: false }).then(() => undefined),
+          submitInferenceServiceResource({ dryRun: false }).then(() => undefined),
+        ];
+        if (!editInfo) {
+          promises.push(
+            createNIMSecret(namespace, NIM_SECRET_NAME, false, false).then(() => undefined),
+            createNIMSecret(namespace, NIM_NGC_SECRET_NAME, true, false).then(() => undefined),
+            createNIMPVC(namespace, nimPVCName, pvcSize, false).then(() => undefined),
+          );
+        } else if (existingPvcSize !== pvcSize && existingPVC && createData) {
+          promises.push(
+            updatePvc(createData, existingPVC, namespace, { dryRun: false }).then(() => undefined),
```

Comment:
Are we sure this is doing anything? I set the size to 32 on create -- and then edited it to 30 -- and the network request doesn't even look like it submitted 30 -- looks to have submitted 32. Perhaps related to my other message.

Reply:
I don't believe it's possible to decrease the size of a PVC.

Reply:
It's not... but it should fail

Reply:
Nvm, it does fail now -- not sure if you changed something here -- but it is working as expected now.